src/app/public/test-utility/test-utility.ts

```diff
@@ -26,4 +30,8 @@ export class SkyAppTestUtility {
     event.initEvent(eventName, settings.bubbles, settings.cancelable);
     element.dispatchEvent(event);
   }
+
+  public static fullSpyOnClass<T>(type: Type<T>): jasmine.SpyObj<T> {
+    return jasmine.createSpyObj(type.name, Object.keys(type.prototype));
```

Comment:
I have a few concerns:

1. The test utility is designed to be framework agnostic so that it would work with Jasmine or any other testing framework.
2. Also, since the method is fairly straight-forward, I'm leaning toward not putting this in the test utility since the complexity doesn't merit a universal implementation, and would probably be better served as a local "helper" function in your SPA/library spec.

Reply:
1. we seem pretty coupled to jasmine, right? are there people testing w/o jasmine? looking through the ADO code i see some use jest, but for e2e [the Zeus docs](https://host.nxt.blackbaud.com/zeus-docs/learn/guides/ui-guidelines#recommended-tools) mention ways to hack a skyux project to run jest, which implies that using something other than jasmine isn't really a paved road. is the thought that we would likely not be using jasmine one day? is that a likely scenario or a edge case? i can make a `skyux-sdk-jasmine-testing` project i suppose, although this would be the only thing in it.
2. it currently is a helper function in local SPAs. as of now, the count is [20 SPAs have this](https://dev.azure.com/blackbaud/Products/_search?action=contents&text=%22function%20fullSpyOnClass%22%20repo%3Askyux*&type=code&lp=code-Project&filters=ProjectFilters%7BProducts%7D). it's copy-pasted in a lot of places. hence why i'm trying to get it in a common spot so ppl can stop copy-pasting it in each project. i dont want to continue that paradigm. if you look at this 20 usages, some of them are slightly wrong (`Spy` vs `SpyObj`) b/c when we made that change we had no centralized place to change. this PR is trying to fix that.

i'm not sure i fully understand the concerns.